### PR TITLE
Fix tolerance and deviation comparison integrate md histo workspace

### LINKED ISSUE #13796: Fix to tolerance comparison in IntegrateMDHistoWorkspace
The comparison of tolerance and deviation of bin-boundaries in `IntegrateMDHistoWorkspace` method `getPrecisionCorrectedCoordinates` does not use the absolute value of deviation.

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Fixes #13796 

**Context**

This fixes an error in the tolerance calcuation. It also addresses a precsion issue that arises when call `JumpToNext()` on an `MDHistoWorkspaceIterator` when the two boxes are essentially equal. 
## For testing

Running the script below takes a sample workspaces and passes it through `IntegrateMDHistoWorkspace`. The Pbins are set to leave everything unchanged. Confirm that the two workspaces are identical. (Note that the signal values show a slight deviation).

The required data file is MDHisto_Osiris. You need to set the path to it in the script. The file can be found here: \olympic\Babylon5\Scratch\Anton\VSI\SAMPLE_WORKSPACES

The script can be found [here](https://gist.github.com/picatess/b17ebd24537edfad2576)

### CODE REVIEW

**Comment by picatess** [comment]
@OwenArnold  The Installer / Disk image files of the Jenkins builds can be found here: \\olympic\Babylon5\Scratch\Anton\Testing\13786_IntegrateMDHistoWorkspace

**Comment by picatess** [comment]
@OwenArnold  Using the minimum of the output box instead of the centre will not work properly, as the calculation of the neighbours (and the corresponding box overlap) would then happen around the min extent -- which is not correct. Please don't test yet.

File: Code/Mantid/Framework/MDAlgorithms/test/IntegrateMDHistoWorkspaceTest.h [hunk 1]
```diff
@@ -7,10 +7,31 @@
 #include "MantidMDAlgorithms/IntegrateMDHistoWorkspace.h"
 #include "MantidDataObjects/MDEventWorkspace.h"
 #include "MantidTestHelpers/MDEventsTestHelper.h"
-
 using Mantid::MDAlgorithms::IntegrateMDHistoWorkspace;
 using namespace Mantid::API;
 
+namespace {
+
+size_t getTheTotalNumberOfBins(IMDHistoWorkspace_sptr ws) {
```
Comment (OwenArnold) [inline Code/Mantid/Framework/MDAlgorithms/test/IntegrateMDHistoWorkspaceTest.h]: Not needed. Just call getNumPoints on the IMDWorkspace

**Comment by OwenArnold** [comment]
In terms of code changes. This looks ok so far.

**Comment by picatess** [comment]
The artefacts which appear when applying `IntegrateMDHistoWorkspace` with Pbin settings which match the original workspace, arise from precision issues when the weight is calculated. The prescision varies on the order of 1e-7. 
* When we are comparing boxes which lie in the same time slice and share a surface this means that the box overlap calculation will have three dimensions for which the overlap comparison should yield 1 and one dimension for which it should yield 0. If we take the rounding issues in to account we end up wiith a signal whcih is on the order of O(1)*O(1)*O(1)*O(10e-7) ~ 10e-7. Which is what  can be seen in the slice viewr
* If the two boxes share only an edge and are in the same time slice then, we expect O(1)*O(1)*O(1e-7)*O(10e-7) ~ 10e-14. These contributions can also be observed in debugging but are cut off by in the slice viewer.

**Pull request merged**